SA/SA.cna

```diff
@@ -101,17 +101,29 @@ sub readbof {
 
 
 alias dir {
-	local('$args $targetdir');
-	$targetdir = iff(-istrue $2, $2, ".\\");
+	local('$params $keys $args $targetdir $subdirs');
 
-	$args = bof_pack($1, "Z", $targetdir);
+	%params = ops(@_);
+	@keys = keys(%params);
+
+	$targetdir = ".\\";
+	$subdirs = 0;
+
+	if ("1" in @keys) {
+		$targetdir = %params["1"];
+		if ("s" in @keys) {
+			$subdirs = 1;
+		}
+	}
```

Comment:
If you move the check for "s" key outside of the if-then clause, you will be able to do a recursive directory listing for the current directory; otherwise, the "/s" option is ignored for the current directory. 
For example, currently "dir /s" is does not do a recursive directory listing of the current directory and is functionally the same as "dir".

Reply:
I want this. Thanks!